web/src/components/drawer-diag.tsx

```
@@ -0,0 +1,76 @@
+        className="mb-4"
+      >
+        {buttonText}
+      </Button>
+      
+      {isOpen && (
+        <div 
+          className="fixed inset-0 z-50 bg-black/50 flex items-center justify-center"
+          onClick={handleClose}
+          onKeyDown={(e) => e.key === 'Enter' && handleClose()}
+          role="dialog"
+          aria-modal="true"
```

Biome (1.9.4)

[error] The elements with the following roles can be changed to the following elements:
<dialog>


For examples and more information, see WAI-ARIA Roles

(lint/a11y/useSemanticElements)